toolchain/check/check.cpp

```diff
@@ -74,46 +77,169 @@ struct UnitInfo {
   llvm::SmallVector<UnitInfo*> incoming_imports;
 };
 
-// Returns the NameId for the entity. May return Invalid for a TODO.
-// TODO: This will need to handle enclosing namespaces.
-static auto GetImportNameId(Parse::NodeId parse_node, Context& context,
-                            const SemIR::File& import_sem_ir,
-                            SemIR::InstId import_inst_id) -> SemIR::NameId {
-  auto import_inst = import_sem_ir.insts().Get(import_inst_id);
-
+// Returns name information for the entity, corresponding to IDs in the import
+// IR rather than the current IR. May return Invalid for a TODO.
+static auto GetImportName(Parse::NodeId parse_node, Context& context,
+                          const SemIR::File& import_sem_ir,
+                          SemIR::Inst import_inst)
+    -> std::pair<SemIR::NameId, SemIR::NameScopeId> {
   switch (import_inst.kind()) {
     case SemIR::InstKind::BindName: {
-      auto bind_name = import_inst.As<SemIR::BindName>();
-      return import_sem_ir.bind_names().Get(bind_name.bind_name_id).name_id;
+      auto bind_name = import_sem_ir.bind_names().Get(
+          import_inst.As<SemIR::BindName>().bind_name_id);
+      return {bind_name.name_id, bind_name.enclosing_scope_id};
     }
 
     case SemIR::InstKind::FunctionDecl: {
-      auto bind_name = import_inst.As<SemIR::FunctionDecl>();
-      return import_sem_ir.functions().Get(bind_name.function_id).name_id;
+      auto function = import_sem_ir.functions().Get(
+          import_inst.As<SemIR::FunctionDecl>().function_id);
+      return {function.name_id, function.enclosing_scope_id};
+    }
+
+    case SemIR::InstKind::Namespace: {
+      auto namespace_inst = import_inst.As<SemIR::Namespace>();
+      auto scope =
```

Comment:
```suggestion
      auto& scope =
```
Accidental `NameScope` copy.